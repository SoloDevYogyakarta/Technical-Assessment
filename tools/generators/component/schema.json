{
  "$schema": "http://json-schema.org/schema",
  "cli": "nx",
  "$id": "component",
  "type": "object",
  "properties": {
    "filename": {
      "type": "string",
      "description": "Write your file name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your filename"
      },
      "$default": {
        "$source": "filename"
      }
    },
    "directoryname": {
      "type": "string",
      "description": "Write your directory name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your directoryname"
      },
      "$default": {
        "$source": "directoryname"
      }
    },
    "projectname": {
      "type": "string",
      "description": "Write your project name ? ",
      "x-prompt": {
        "type": "string",
        "message": "Write your projectname"
      },
      "$default": {
        "$source": "projectname"
      }
    }
  }
}
