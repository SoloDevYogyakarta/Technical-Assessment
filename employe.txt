“username”: string, “firstName”:string, “lastName”:string, “email”:string, “birthDate”:datetime, “basicSalary”:double, “status”:string, “group”:string, “description”:datetime