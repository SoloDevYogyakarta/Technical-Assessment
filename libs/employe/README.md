# employe

This library was generated with [Nx](https://nx.dev).

## Running unit tests

Run `nx test employe` to execute the unit tests.
