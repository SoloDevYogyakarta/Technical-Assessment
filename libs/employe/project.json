{
  "name": "employe",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/employe/src",
  "prefix": "nov9-technical-assessment",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/employe/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/employe/**/*.ts", "libs/employe/**/*.html"]
      }
    }
  },
  "tags": []
}
