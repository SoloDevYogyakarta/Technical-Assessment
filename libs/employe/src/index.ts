export * from './lib/employe.module';
