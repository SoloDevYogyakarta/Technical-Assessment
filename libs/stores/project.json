{
  "name": "stores",
  "$schema": "../../node_modules/nx/schemas/project-schema.json",
  "projectType": "library",
  "sourceRoot": "libs/stores/src",
  "prefix": "nov9-technical-assessment",
  "targets": {
    "test": {
      "executor": "@nrwl/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/stores/jest.config.ts",
        "passWithNoTests": true
      }
    },
    "lint": {
      "executor": "@nrwl/linter:eslint",
      "options": {
        "lintFilePatterns": ["libs/stores/**/*.ts", "libs/stores/**/*.html"]
      }
    }
  },
  "tags": []
}
