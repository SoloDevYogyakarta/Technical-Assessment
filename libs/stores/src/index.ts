export * from './lib/stores.module';
export * from './lib/constants/employe-type/employe-type';
export * from './lib/selectors/employe-selector/employe-selector';
export * from './lib/actions/employe-action/employe-action';
export * from './lib/reducers/employe-reducer/employe-reducer';
