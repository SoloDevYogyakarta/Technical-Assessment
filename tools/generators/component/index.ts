import {
  Tree,
  formatFiles,
  getProjects,
  names,
  generateFiles,
} from '@nrwl/devkit';
import { join } from 'path';
import { ISchema } from '../schema';

export default async function (tree: Tree, schema: ISchema) {
  const workspace = getProjects(tree).get(schema.projectname);
  const templatePath = 'tools/generators/component/templates';
  const directoryPath = schema.directoryname
    ? `${workspace?.sourceRoot}/lib/${schema.directoryname}`
    : `${workspace?.sourceRoot}/lib`;
  const filename = `${schema.filename}.component`;
  const targetPath = join(directoryPath, filename);
  const componentNames = names(filename);
  const substitutions = {
    tmpl: '',
    ...componentNames,
  };
  generateFiles(tree, templatePath, targetPath, substitutions);
  await formatFiles(tree);
}
