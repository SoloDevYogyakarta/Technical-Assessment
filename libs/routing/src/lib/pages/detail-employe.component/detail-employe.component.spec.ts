import { DetailEmployeComponent } from './detail-employe.component';
import {
  TestBed,
  TestBedStatic,
  ComponentFixture,
} from '@angular/core/testing';
import { StoreModule } from '@ngrx/store';
import { RouterTestingModule } from '@angular/router/testing';
import { employeReducer } from '@nov9-technical-assessment/stores';
import { routes } from '../../route.module/route.module';
import { EmployeComponent } from '../employe.component/employe.component';
import { NgHeroiconsModule } from '@dimaslz/ng-heroicons';

describe('DetailEmployeComponent', () => {
  let container: TestBedStatic;
  let component: ComponentFixture<DetailEmployeComponent>;

  beforeEach(async () => {
    container = await TestBed.configureTestingModule({
      imports: [
        StoreModule.forRoot([employeReducer]),
        RouterTestingModule.withRoutes(routes),
        NgHeroiconsModule,
      ],
      declarations: [DetailEmployeComponent, EmployeComponent],
    }).compileComponents();

    component = TestBed.createComponent(DetailEmployeComponent);
    component.detectChanges();
  });

  it('should render snapshot', () => {
    expect(component).toMatchSnapshot();
  });

  it("should have as title 'DetailEmployeComponent'", () => {
    const template = component.componentInstance;
    expect(template.title).toEqual('DetailEmployeComponent');
  });

  it('should render title', () => {
    const template = component.nativeElement as HTMLElement;
    expect(template.querySelector('div')?.textContent).toContain('');
  });

  afterEach(async () => {
    await jest.clearAllMocks();
  });
});
