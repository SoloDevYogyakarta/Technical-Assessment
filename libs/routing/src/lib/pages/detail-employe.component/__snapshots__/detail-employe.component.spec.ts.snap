// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DetailEmployeComponent should render snapshot 1`] = `
<detailemploye-component
  __ngContext__="0"
  employe$={[Function Store]}
  isLoading$={[Function Store]}
  route={[Function ActivatedRoute]}
  router={[Function Router]}
  store={[Function Store]}
  title={[Function String]}
>
  <div
    class="w-full px-4"
  >
    <div
      class="w-full bg-gray-100 p-3 flex items-center justify-end rounded-md my-4"
    >
      <arrow-right-solid-icon
        class="cursor-pointer"
      >
        <svg
          aria-hidden="true"
          fill="currentColor"
          style="width: 24px; height: 24px;"
          viewBox="0 0 24 24"
          xmlns="http://www.w3.org/2000/svg"
        >
          <path
            clip-rule="evenodd"
            d="M12.97 3.97a.75.75 0 0 1 1.06 0l7.5 7.5a.75.75 0 0 1 0 1.06l-7.5 7.5a.75.75 0 1 1-1.06-1.06l6.22-6.22H3a.75.75 0 0 1 0-1.5h16.19l-6.22-6.22a.75.75 0 0 1 0-1.06z"
            fill-rule="evenodd"
          />
        </svg>
      </arrow-right-solid-icon>
    </div>
  </div><div
    class="flex items-center justify-center w-full flex-wrap"
  >
    <div
      class="w-full max-w-[330px] bg-white shadow-md rounded-lg m-4 h-96 overflow-hidden"
    >
      <div
        class="w-full h-44 bg-gray-200"
      >
        <img
          alt=""
          class="w-full h-full bg-center bg-cover bg-no-repeat"
          src="https://api-prod-minimal-v4.vercel.app/assets/images/covers/cover_6.jpg"
        />
      </div>
      <div
        class="relative"
      >
        <div
          class="w-24 h-24 bg-white shadow-md rounded-full absolute top-[-50px] left-[36%] overflow-hidden"
        >
          <img
            alt=""
            class="bg-center bg-cover bg-no-repeat"
            src="https://api-prod-minimal-v4.vercel.app/assets/images/avatars/avatar_5.jpg"
          />
        </div>
      </div>
      <div
        class="w-full pt-12"
      >
        <p
          class="text-center text-gray-600 font-semibold"
        >
          <a
            class="transition duration-500 hover:text-blue-500 cursor-pointer"
          >
          </a>
        </p>
        <p
          class="text-center text-gray-400 font-semibold text-xs"
        >
        </p>
        <div
          class="text-center text-green-400 font-semibold text-sm"
        >
        </div>
        <p
          class="text-center text-gray-500 font-semibold text-sm"
        >
           Salery :  
        </p>
        <div
          class="p-2"
        >
          <button
            class="w-full bg-red-500 p-2 rounded-lg flex items-center justify-center"
          >
            <trash-solid-icon
              class="text-white"
            >
              <svg
                aria-hidden="true"
                fill="currentColor"
                style="width: 24px; height: 24px;"
                viewBox="0 0 24 24"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  clip-rule="evenodd"
                  d="M16.5 4.478v.227a48.816 48.816 0 0 1 3.878.512.75.75 0 1 1-.256 1.478l-.209-.035-1.005 13.07a3 3 0 0 1-2.991 2.77H8.084a3 3 0 0 1-2.991-2.77L4.087 6.66l-.209.035a.75.75 0 0 1-.256-1.478A48.567 48.567 0 0 1 7.5 4.705v-.227c0-1.564 1.213-2.9 2.816-2.951a52.662 52.662 0 0 1 3.369 0c1.603.051 2.815 1.387 2.815 2.951zm-6.136-1.452a51.196 51.196 0 0 1 3.273 0C14.39 3.05 15 3.684 15 4.478v.113a49.488 49.488 0 0 0-6 0v-.113c0-.794.609-1.428 1.364-1.452zm-.355 5.945a.75.75 0 1 0-1.5.058l.347 9a.75.75 0 1 0 1.499-.058l-.346-9zm5.48.058a.75.75 0 1 0-1.498-.058l-.347 9a.75.75 0 0 0 1.5.058l.345-9z"
                  fill-rule="evenodd"
                />
              </svg>
            </trash-solid-icon>
            <span
              class="text-sm font-bold text-white"
            >
              Delete
            </span>
          </button>
        </div>
      </div>
    </div>
  </div>
</detailemploye-component>
`;
